core/templates/components/common-layout-directives/navigation-bars/side-navigation-bar.component.spec.ts

```diff
@@ -83,13 +109,92 @@ describe('Side Navigation Bar Component', () => {
     expect(componentInstance.currentUrl).toEqual(currentUrl);
   });
 
-  it('should toggle classroom submenu', () => {
-    componentInstance.classroomSubmenuIsShown = false;
-    componentInstance.toggleClassroomSubmenu();
-    expect(componentInstance.classroomSubmenuIsShown).toBeTrue();
+  it('should able to stop click propagation further', () => {
+    const clickEvent = new CustomEvent('click');
+    spyOn(clickEvent, 'stopPropagation');
+    componentInstance.stopclickfurther(clickEvent);
+  });
+
+  it('should toggle learn submenu', () => {
+    componentInstance.learnSubmenuIsShown = false;
+    componentInstance.togglelearnSubmenu();
+    expect(componentInstance.learnSubmenuIsShown).toBeTrue();
+    componentInstance.togglelearnSubmenu();
+    expect(componentInstance.learnSubmenuIsShown).toBeFalse();
+  });
+
+  it('should toggle get involved submenu', () => {
+    componentInstance.getinvolvedSubmenuIsShown = false;
+    componentInstance.togglegetinvolvedSubmenu();
+    expect(componentInstance.getinvolvedSubmenuIsShown).toBeTrue();
+    componentInstance.togglegetinvolvedSubmenu();
+    expect(componentInstance.getinvolvedSubmenuIsShown).toBeFalse();
   });
 
   it('should get static image url', () => {
     expect(componentInstance.getStaticImageUrl('test')).toEqual(imageUrl);
   });
+
+  it('should close sidebar on swipe left', () => {
+    spyOn(sidebarStatusService, 'closeSidebar');
+    componentInstance.closeSidebarOnSwipeleft();
+    expect(sidebarStatusService.closeSidebar).toHaveBeenCalled();
+  });
+
+  it('should navigate to classroom page when user clicks on' +
+  '\'Basic Mathematics\'', fakeAsync(() => {
+    expect(mockWindowRef.nativeWindow.location.href).toBe('');
+
+    componentInstance.navigateToClassroomPage('/classroom/url');
+    tick(151);
+
+    expect(mockWindowRef.nativeWindow.location.href).toBe('/classroom/url');
+  }));
+
+  it('should registers classroom header click event when user clicks' +
+  ' on \'Basic Mathematics\'', () => {
+    spyOn(siteAnalyticsService, 'registerClassroomHeaderClickEvent');
+
+    componentInstance.navigateToClassroomPage('/classroom/url');
+
+    expect(siteAnalyticsService.registerClassroomHeaderClickEvent)
+      .toHaveBeenCalled();
+  });
+
+  it('should check if ngoninit work as expected', fakeAsync(() => {
```

Comment:
something like `should populate properties on component initialization` would be better

Reply:
done!